### DISCUSSION TITLE
Building MtoH for Maya 2023

### DISCUSSION THREAD

**Discussion by Vsevolod1983** [Autodesk/maya-usd#2256] [Q&A] [answered]
Hello!
Since Maya 2023 is shipped and contains Maya USD plugin I assume that I can build MtoH from sources using maya 2023 sdk.

I tried to build with Python 3.7 as mentioned in build documentation. However I have loading error which says about conflicting python versions.
Maya 2023 has Python 3.9 inside. 

Does it mean I should build everything including PixarUSD with Python 3.9 ?
Any other suggestions about building for 2023 ? Cmake version etc ?
Thank you!

**Comment by santosg87** [comment, marked as answer]
@Vsevolod1983 it seems like i you might have got the answers you need on #2264. are you ok with me closing this discussion thread?

**Comment by seando-adsk** [comment]
@Vsevolod1983 FYI, you should not need to build your own USD. Inside of the MayaUsd plugin we have our custom built USD binaries. And you'll find a .zip file containing all the headers and libs needed to build MayaUsd. Just extract this .zip if you want to build your own USD.

**Comment by Vsevolod1983** [comment]
@seando-adsk 
Hello!
What if we need Pixar USD to be built with --openimageio and --materialx flags ? I assume that you build USD with --materialx flag but without --openimageio?

Is this correct ?

**Reply by seando-adsk** [reply]
No, we do build USD with both openimage and materialx. We also build with OpenExr (statically), Hdf5 (statically) and Alembic (statically) to avoid any conflicts with versions in Maya.

When building USD we also use TBB and OpenSubdiv from Maya to avoid any potential conflicts.

However you should be able to build and use your own USD.

Sean
